fixme-parent/fixme-broker/src/main/java/za/co/wethinkcode/broker/Broker.java

```diff
@@ -1,35 +1,122 @@
 package za.co.wethinkcode.broker;
 
+import lombok.Data;
+import za.co.wethinkcode.ExecutionReportDecoded;
+import za.co.wethinkcode.FixMessage;
+import za.co.wethinkcode.exceptions.FixMessageException;
+
 import java.io.IOException;
 import java.io.PrintWriter;
 import java.net.Socket;
 import java.net.UnknownHostException;
 import java.util.Scanner;
 
+@Data
 public class Broker {
-	
-	// TODO maybe maintain one instance of the Scanner and PrintWriter
-	
-	private String brokerId;
+
+    // TODO maybe maintain one instance of the Scanner and PrintWriter
+
+    private String brokerId;
     private final int PORT = 5000;
     private final String HOST = "127.0.0.1";
-    Socket socket;
-    
-    public Broker() throws UnknownHostException, IOException{
-    	socket = new Socket(HOST, PORT);
-    	System.out.println("Connection to router has been established");
-    	brokerId = processResponse();
-    	System.out.println("Allocated ID: " + brokerId);    	
+    private Socket socket;
+    private Scanner fromRouter;
```

Comment:
This was added as processResponse was creating its own new scanner, thus dropping the state of the scanner created in the constructor, which will contain messages if the broker connects first and then the market.